corehq/apps/locations/models.py

```diff
@@ -717,13 +717,16 @@ def location_type(self):
 
     @location_type.setter
     def location_type(self, value):
+        self.set_location_type(value)
+
+    def set_location_type(self, location_type_name):
```

Comment:
I'm not sure, because you're not just setting the name, you're also setting the object by passing in the name. No strong feelings.